lms/djangoapps/certificates/views.py

```diff
@@ -121,3 +132,77 @@ def update_certificate(request):
+    If a certificate is not available, we display a "Sorry!" screen instead
+    """
+    invalid_template_path = 'certificates/invalid.html'
+
+    # Feature Flag check
+    if not settings.FEATURES.get('CERTIFICATES_HTML_VIEW', False):
+        return render_to_response(invalid_template_path)
+
+    context = {}
+    course_id = request.GET.get('course', None)
+    context['course'] = course_id
+    if not course_id:
+        return render_to_response(invalid_template_path, context)
+
+    # Course Lookup
+    try:
+        course_key = CourseKey.from_string(course_id)
+    except InvalidKeyError:
+        return render_to_response(invalid_template_path, context)
+    course = modulestore().get_course(course_key)
+    if not course:
+        return render_to_response(invalid_template_path, context)
+
+    # Certificate Lookup
+    try:
+        certificate = GeneratedCertificate.objects.get(
+            user=request.user,
+            course_id=course_key
+        )
+    except GeneratedCertificate.DoesNotExist:
+        return render_to_response(invalid_template_path, context)
+
+    # Load static output values from configuration,
+    configuration = CertificateHtmlViewConfiguration.get_config()
+    context = configuration.get('default', {})
+    # Override the defaults with any mode-specific static values
+    context.update(configuration.get(certificate.mode, {}))
+    # Override further with any course-specific static values
+    context.update(course.cert_html_view_overrides)
+
+
+    # Populate dynamic output values using the course/certificate data loaded above
+    user_fullname = request.user.profile.name
+    company_name = context.get('company_name')
+    context['accomplishment_copy_name'] = user_fullname
+    accd = 'a course of study offered by <span class="detail--xuniversity">{0}</span>'.format(course.org)
+    accd = accd + ', through <span class="detail--company">{0}</span>.'.format(company_name)
+    context['accomplishment_copy_course_description'] = accd
+    context['accomplishment_copy_course_org'] = course.org
+    context['accomplishment_copy_course_name'] = course.display_name
+    context['accomplishment_more_title'] = "More Information About {0}'s Certificate:".format(user_fullname)
+    context['certificate_date_issued'] = certificate.modified_date.strftime("%B %m, %Y")
+    context['certificate_id_number'] = certificate.verify_uuid
+    context['certificate_verify_url'] = "{0}{1}{2}".format(
+        context.get('certificate_verify_url_prefix'),
+        certificate.verify_uuid,
+        context.get('certificate_verify_url_suffix')
+    )
+    context['copyright_text'] = "&copy; {0} {1}. All rights reserved".format(datetime.now().year, company_name)
+    dmd = "This is a valid {0} certificate for {1}, ".format(company_name, user_fullname)
+    dmd = dmd + "who participated in {0} {1}".format(course.org, course.number).format()
```

Comment:
You can do this for i18n:
```python
string_1 = _(
    "This is a string that includes variables that are readable "
    "such as {platform_name} and {date}, rather than inscrutable "
    "positional variables."
).format(
    platform_name=settings.get(platform_name),
    date=date.today
)
```